crates/bevy_reflect/src/map.rs

```diff
@@ -673,4 +724,14 @@ mod tests {
             assert_eq!(size, iter.index);
         }
     }
+
+    #[test]
+    fn remove() {
+        let mut map = DynamicMap::default();
+        map.insert(0, 0);
+        map.insert(1, 1);
+
+        map.remove(&0);
+        assert!(map.get(&1).is_some());
```

Comment:
Nit: We could probably add a couple more assertions to verify that removing the last element and a non-existent one is also working:
```suggestion
        assert!(map.remove(&0).is_some());
        assert!(map.get(&0).is_none());
        assert!(map.get(&1).is_some());
        
        assert!(map.remove(&1).is_some());
        assert!(map.get(&1).is_none());
        
        assert!(map.remove(&1).is_none());
        assert!(map.get(&1).is_none());
```